lib/matplotlib/axes/_axes.py

```diff
@@ -4559,6 +4559,10 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
             collection.set_cmap(cmap)
             collection.set_norm(norm)
             collection._scale_norm(norm, vmin, vmax)
+        elif any({cmap, norm, vmin, vmax}):
+            _api.warn_external(
```

Comment:
It might be worth adding a bit of logic to tell the user which one they passed?

Something like:

```
extra_keys = {k for k, v in {'cmap': cmap, 'norm': norm, 'vmin': vmin, 'vmax': vmax} if v is not None}
```

and then include that in the error message?